### PR TITLE
feat(@angular-devkit/build-optimizer): strip Angular decorators in __…

### PR DESCRIPTION
**Pull request by filipesilva**
…decorate call

### CODE REVIEW

File: packages/angular_devkit/build_optimizer/src/transforms/scrub-file.ts [hunk 1]
```diff
@@ -261,6 +309,36 @@ function pickDecorationNodesToRemove(
...
+  const ngDecoratorCalls = elements.filter((el) => {
+    if (el.expression.kind !== ts.SyntaxKind.Identifier) {
+      return false;
+    }
+    const id = el.expression as ts.Identifier;
+
+    return identifierIsMetadata(id, ngMetadata, checker);
+  });
+
+  return (elements.length > ngDecoratorCalls.length) ? ngDecoratorCalls : [exprStmt];
```
Comment (IgorMinar) [inline packages/angular_devkit/build_optimizer/src/transforms/scrub-file.ts]: this logic is hard to follow. maybe you want to say:

```
// if all decorators are metadata decorators then return the whole `Class = __decorate([...])'` statement so that it is removed in entirety
return (elements.length === ngDecoratorCalls.length) ? [exprStmt] : ngDecoratorCalls;
```

Comment (filipesilva) [reply packages/angular_devkit/build_optimizer/src/transforms/scrub-file.ts]: Fixed
